FHICORC.Backend/FHICORC.Tests.UnitTests/DGCGComponentTests/DGCGRevocationServiceComponentTests.cs

```diff
@@ -63,6 +66,69 @@ public void Setup()
             }
         }
 
+        private (List<string> HashCollectionOne, List<string> HashCollectionTwo) GetRandomHashesInLists(int amount)
+        {
+            var hashList = new List<string>();
+            var hashList2 = new List<string>();
+            for (int i = 0; i < amount; i++)
+            {
+                hashList.Add(i.ToString());
+                hashList2.Add((i+10000).ToString());
+            }
+            return (hashList, hashList2);
+        }
+
+        [TestCase (2999, 1, 3, 3)]
+        public void UploadHashesTest(int amount, int hashCount1, int batchCount1, int batchCount2)
+        {
+            // Arrange
+            (var list1, var list2) = GetRandomHashesInLists(amount);
+            var countBatchesBefore = _coronapassContext.RevocationBatch.Count();
+            var countHashesBefore = _coronapassContext.RevocationHash.Count();
+
+            // Act
+            _revocationService.UploadHashes(list1);
+
+            // Assert
+            Assert.AreEqual(_coronapassContext.RevocationBatch.Count(), countBatchesBefore + batchCount1);
+            Assert.AreEqual(_coronapassContext.RevocationHash.Count(), countHashesBefore + amount);
+
+            // Arrange
+            var countBatchesAfter = _coronapassContext.RevocationBatch.Count();
```

Comment:
This can be moved to before previous assert and used in the assert, renamed as batchCountAfterUpload.

It should read "Assert.AreEqual(batchCountAfterUpload , batchCountPreUpload + batchCount1);"

And if batchCount1 has some better description why its 3 then change that